web3/providers/websocket/request_processor.py

```diff
@@ -246,12 +250,53 @@ def pop_raw_response(
             if raw_response is not None:
                 self._provider.logger.debug(
                     "Cached response popped from cache to be processed:\n"
+                    f"    response_id={raw_response.get('id')},\n"
                     f"    cache_key={cache_key},\n"
-                    f"    raw_response={raw_response}"
+                    f"    raw_response={raw_response}",
                 )
 
         return raw_response
 
+    async def _await_next_cached_response(
+        self,
+        request_id: Optional[RPCId] = None,
+        subscription: bool = False,
+    ) -> RPCResponse:
+        """
+        Wait for the cached response, by type, while caching all other responses.
+
+        - If subscription is True, wait for the next cached subscription response.
+        - If subscription is False, wait for the next cached response that matches the
+            ``request_id``.
+        """
+        if request_id is None and not subscription:
+            raise ValueError("Must provide a request_id when subscription is False.")
+        elif subscription and request_id is not None:
+            raise ValueError(
+                "``request_id`` must be ``None`` when ``subscription`` is True."
```

Comment:
Nit: Could include the value of `request_id` in the error message.

Reply:
This is an internal method to us and I validated it just so we know we shouldn't use it in our code under certain conditions. If it isn't `None`, it shouldn't be _anything_, do you think it would still be nice to have it? 👀

Reply:
That's good to know. The logic implies the value is only relevant in the case that `subscription` is false. Thought about including a comment in the block at the top of this method but even that seems redundant. Should be ok to leave as is!